lib.js

```diff
@@ -99,8 +102,27 @@ module.exports = (options) => {
         });
     };
 
+    const hasValidCertificate = (pagesDomain) => {
+        if (options.domain.includes(pagesDomain.domain)) {
```

Comment:
This function should not care about `option.domain`,  it should only care about the domain passed as a parameter. If we need to filter other domains it should be done at the call site instead of here.